src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
+    }
+    break;
+
+    case ZCL_ROUTING_ROLE_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(otDeviceRole);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        otDeviceRole role = otThreadGetDeviceRole(mOTInst);
+        Encoding::LittleEndian::Put16(*buffer, role);
+    }
+    break;
+
+    case ZCL_NETWORK_NAME_ATTRIBUTE_ID: {
+        type       = kTLVType_UTF8String;
+        ReadLength = OT_NETWORK_NAME_MAX_SIZE + 1;
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        ChipLogProgress(DeviceLayer, "Read NetworkName= %s", otThreadGetNetworkName(mOTInst));
+        strncpy(reinterpret_cast<char *>(*buffer), otThreadGetNetworkName(mOTInst), ReadLength);
+        ChipLogProgress(DeviceLayer, "Copied NetworkName= %s", *buffer);
+    }
+    break;
+
+    case ZCL_DIAG_PAN_ID_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(otPanId);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        otPanId panId = otLinkGetPanId(mOTInst);
+        Encoding::LittleEndian::Put16(*buffer, panId);
+    }
+    break;
+
+    case ZCL_DIAG_EXTENDED_PAN_ID_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint64_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        const otExtendedPanId * pExtendedPanid = otThreadGetExtendedPanId(mOTInst);
+        Encoding::LittleEndian::Put64(*buffer, Encoding::BigEndian::Get64(pExtendedPanid->m8));
+    }
+    break;
+
+    case ZCL_MESH_LOCAL_PREFIX_ATTRIBUTE_ID: {
+        const uint8_t kOctStrPrefixLen =
+            (OT_MESH_LOCAL_PREFIX_SIZE + 1) * 2 + 1; // For each hex character of Len + Prefix + NULL char
+        type       = kTLVType_ByteString;
+        ReadLength = kOctStrPrefixLen;
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        const otMeshLocalPrefix * pMeshLocalPrefix                 = otThreadGetMeshLocalPrefix(mOTInst);
+        uint8_t formattedIpv6Prefix[OT_MESH_LOCAL_PREFIX_SIZE + 1] = { 0 };
+        formattedIpv6Prefix[0]                                     = OT_IP6_PREFIX_BITSIZE;
+        memcpy(&formattedIpv6Prefix[1], pMeshLocalPrefix->m8, OT_MESH_LOCAL_PREFIX_SIZE);
+
+        chip::Encoding::BytesToUppercaseHexString(formattedIpv6Prefix, OT_MESH_LOCAL_PREFIX_SIZE + 1,
+                                                  reinterpret_cast<char *>(*buffer), ReadLength);
```

Comment:
Why is there a hex string here?  ByteString just means bytes, no?

Reply:
The spec specify this attributes as an octet string and give the following example
Preferred minimal encoding: Prefix 2001:0DB8:BB00::/40 → 6 octets → 2820010DB8BB
using BytesToUppercaseHexString produce the right output with tlv type `kTLVType_ByteString`

Here is an example of the ouput
`AttributeReadResult(path=AttributePath(nodeId=135, endpointId=0, clusterId=53, attributeId=5), status=0, value=b'40FDDEAD00BEEF0000\x00')`

Did I misunderstood the type oct string and it should rather be a byte array ?

Reply:
octet string in the spec is a byte array, yes.